### ISSUE TITLE
Layout can't be located in folder with routing v2

### ISSUE THREAD

**Issue by ZiiMakc** [remix-run/remix#5261]
### What version of Remix are you using?

1.11.1

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

npx create-remix@latest

Switch to V2 routing, and run `remix routes`

### Expected Behavior

Following layout should work no matter if files or the same folders with `index.tsx` inside are used:

`user.$id/index.tsx` - user layout
`user.$id._index/index.tsx` - nested in user layout
```
    <Route path="user/:id" file="routes/user.$id.tsx">
      <Route index file="routes/user.$id._index.tsx" />>
    </Route>
```

### Actual Behavior

Works:

`user.$id.tsx` - user layout
`user.$id._index.tsx` - nested in user layout
```
    <Route path="user/:id" file="routes/user.$id.tsx">
      <Route index file="routes/user.$id._index.tsx" />>
    </Route>
```

Doesn't: 

`user.$id/index.tsx` - user layout
`user.$id._index/index.tsx` - nested in user layout

> Error: Path "/user/:id" defined by route "user.$id._index/index" conflicts with route "user.$id/index"

**Comment by ZiiMakc** [comment]
After some testing I can say that `user.$id._index/index.tsx` can be located in folder, but not it's layout `user.$id/index.tsx` that needs to be a file `user.$id.tsx`

**Comment by MichaelDeBoey** [comment]
Closed by #5228
